swimlane/core/adapters/record.py

```diff
@@ -194,34 +194,109 @@ def create_batch(self, *records):
 
+        Keyword Args:
+            values (dict): Dictionary of one or more 'field_name': 'new_value' pairs to update
+
 
         Examples:
 
             ::
 
-                # Bulk delete records by filter
+                # Bulk update records by filter
 
-                app.records.bulk_delete(('Field_1', 'equals', value1),
-                                        ('Field_2', 'equals', value2))
+                app.records.bulk_modify(('Field_1', 'equals', value1),
+                                        ('Field_2', 'equals', value2),
+                                        values={
+                                                "Field3": value3,
+                                                "Field_4": value4
+                                                })
 
-                # Bulk delete records
+                # Bulk update records
 
                 record1 = app.records.get(tracking_id='APP-1')
                 record2 = app.records.get(tracking_id='APP-2')
                 record3 = app.records.get(tracking_id='APP-3')
 
-                app.records.bulk_delete(record1, record2, record3)
+                app.records.bulk_modify(record1, record2, record3, values={"Field_Name": new_value})
+
+
 
+                """
+        values = kwargs.pop('values', None)
 
+        if kwargs:
+            raise ValueError('Unexpected arguments: {}'.format(kwargs))
+
+        if not values:
+            raise ValueError('Must provide "values" as keyword argument')
+
+        if not isinstance(values, dict):
+            raise ValueError("values parameter must be dict of {'field_name': 'update_value'} pairs")
+
+        validate_filters_or_records(filters_or_records)
+
+        data_dict = {}
+        record_stub = record_factory(self._app)
+
+        # build record_id list
+        if isinstance(filters_or_records[0], Record):
+            record_ids = []
+            for record in filters_or_records:
+                record_ids.append(record.id)
+            data_dict['recordIds'] = record_ids
+
+        # build filters
+        elif isinstance(filters_or_records[0], tuple):
+            filters = []
+            for filter_tuples in filters_or_records:
+                field = record_stub.get_field(filter_tuples[0])
+                filters.append({
+                    "fieldId": field.id,
+                    "filterType": filter_tuples[1],
+                    "value": field.get_report(filter_tuples[2])
+                })
+            data_dict['filters'] = filters
+
+        # build modifications
+        modifications = []
+        for field_name, update_value in values.items():
+            mod_field = record_stub.get_field(field_name)
+            modifications.append({
+                "fieldId": {
+                    "value": mod_field.id,
+                    "type": "Id"
+                },
+                "value": mod_field.get_report(update_value),
```

Comment:
Did you test that `.get_report` is what the server's expecting vs. `.get_swimlane`?